doc/website/blog: erd and json blog (#1331)
ariga/atlas: doc/website/blog/2022-12-06-erd-and-json.md

---
title: "Exploring Database Schemas with Atlas"
authors: rotemtam
tags: [inspection, visualization, json, erd]
image: https://atlasgo.io/uploads/images/explore_twitter_card.png
---

> Atlas is a most commonly used for managing and applying schema changes to databases, but it can also be used for

Reviewer: ```suggestion
Atlas is most commonly used for managing and applying schema changes to databases, but it can also be used for
```

> something else: exploring and understanding the database schemas.

Reviewer: ```suggestion
something else: exploring and understanding database schemas.
```

With inspection, Atlas connects to your database, analyzes its structure from the metadata tables, and
creates a graph data structure that maps all the entities and relations within the database. Atlas can then
take this graph and represent it in various formats for users to consume. In this post, I will present two such

> forms of representation: Entity Relationship Diagrams (ERD) and a JSON documents.

Reviewer: ```suggestion
forms of representation: Entity Relationship Diagrams (ERDs) and JSON documents.
```

## Schemas as ERDs

> One of the most useful ways to represent a database schema is using an Entity-relation Diagram (ERD). This allows

Reviewer: ```suggestion
One of the most useful ways to represent a database schema is using an Entity Relationship Diagram (ERD). This allows
```

developers to see the schema in a visual and intuitive way, making it easy to understand the relationships between

> different elements of your database. When using ERDs, because the data is presented in a graph format, you can easily

Reviewer: ```suggestion
different elements of the database. When using ERDs, because the data is presented in a graph format, you can easily
```

   ```

## Schemas as JSON documents

In addition to producing ERDs, Atlas can also produce a JSON document that represents the database schema.
One of the key benefits of representing the database schema as a JSON document is that it allows you to use standard
tools like `jq` to analyze the schema programmatically. `jq` is a popular command-line tool for working with JSON data,
and it can be especially useful for exploring and manipulating the schema data generated by Atlas.

With jq, you can easily extract specific information from the schema, such as the names of all the tables in the
database or the foreign key relationships between different entities. This makes it easy to write scripts or programs
that can automatically analyze the schema and identify potential issues or opportunities for optimization.

To get the JSON representation of your database schema, you can use the `atlas schema inspect` command with a
custom logging format:

```text
atlas schema inspect -u '<url>' --log '{{ json . }}'
```

Once your schema is represented as a JSON document, you can use `jq` to analyze it. For example, to get a list of all

> the tables that contain a foreign key:

Reviewer: ```suggestion
the tables that contain a foreign key, run:
```
